### PR TITLE
Fix creation of different field types

### PR DESCRIPTION
**Pull request by amitaibu**
Also split preprocess per field type methods, for POST - this would make it easier to extend according to field types.

### CODE REVIEW

File: plugins/restful/RestfulEntityBase.php [hunk 1]
```diff
@@ -926,58 +926,130 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $request, $
...
+   * @param array $field_info
+   *   The field info array.
+   *
+   * @return mix
+   *   The value to set using the wrapped property.
+   */
+  protected function propertyValuesPreprocessFile($property_name, $value, $field_info) {
+    if ($field_info['cardinality'] == 1) {
+      // Single value.
+      return array('fid' => $value);
```
Comment (snirs90) [inline plugins/restful/RestfulEntityBase.php]: As far as I've checked and tested, Need to add "display" property.
`display => 1`
if it's not set, it's throws an SQL error:
`PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'field_file_display' cannot be null`

Or maybe I'm doing something wrong? (tried without display property):

``` php
$wrapper = entity_metadata_wrapper('node', 1);
$wrapper->field_file->set(array('fid' => 2));
$wrapper->save();
```

File: plugins/restful/RestfulEntityBase.php [hunk 2]
```diff
@@ -926,58 +926,130 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $request, $
...
     }
 
-    // Return the value as is.
-    return $value;
+    $return = array();
+    foreach ($value as $delta => $single_value) {
+      $return[$delta]['fid'] = $single_value;
```
Comment (snirs90) [inline plugins/restful/RestfulEntityBase.php]: Same here about the display property:
`$return[$delta]['display'] = 1;`

File: plugins/restful/RestfulEntityBase.php [hunk 3]
```diff
@@ -926,58 +926,130 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $request, $
   /**
    * Massage the value to set according to the format expected by the wrapper.
    *
-   * @param $property_name
+   * @param string  $property_name
```
Comment (snirs90) [inline plugins/restful/RestfulEntityBase.php]: One space too much after `string`

File: plugins/restful/RestfulEntityBase.php [hunk 4]
```diff
@@ -926,58 +926,130 @@ protected function setPropertyValues(EntityMetadataWrapper $wrapper, $request, $
...
+  protected function propertyValuesPreprocessFile($property_name, $value, $field_info) {
+    if ($field_info['cardinality'] == 1) {
+      // Single value.
+      return array('fid' => $value);
     }
 
-    // Return the value as is.
-    return $value;
+    $return = array();
+    foreach ($value as $delta => $single_value) {
```
Comment (snirs90) [inline plugins/restful/RestfulEntityBase.php:1057]: I think we should add `explode()` to commas same as in entity-reference case if we want to pass few file IDs with comma separated through a POST request.

**Comment by amitaibu** [comment]
POST now works (+tested) for text fields (wit and without text-processing), entity reference, taxonomy term reference, file field and image field.

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch ^^

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**